bugsnag-android-core/src/main/java/com/bugsnag/android/NativeInterface.java

```diff
@@ -365,13 +365,24 @@ public static void notify(@NonNull final String name,
             public boolean onError(@NonNull Event event) {
                 event.updateSeverityInternal(severity);
                 List<Error> errors = event.getErrors();
+                Error error = event.getErrors().get(0);
+
+                // update the thread's stacktrace and type to C
+                for (Thread thread : event.getThreads()) {
+                    if (thread.getErrorReportingThread()) {
+                        thread.setStacktrace(error.getStacktrace());
```

Comment:
Why do we want to update the stacktrace, not just the type?

Reply:
After further consideration I don't think is required - we only alter the thread's stacktrace for unhandled errors. Reverting this change.